modules/cudf/src/series/categorical.ts

```diff
@@ -166,35 +169,37 @@ export class CategoricalSeries<T extends DataType> extends Series<Categorical<T>
     const cur_df = new DataFrame(new ColumnAccessor({old_codes: cur_codes, cur_index: cur_index}));
     const new_df = new DataFrame(new ColumnAccessor({new_codes: new_codes, categories: new_cats}));
 
-    //
     // 1. Join the old and new categories to align their codes
+    const tmp1 = old_df.join({
+      how: 'left',
+      on: ['categories'],
+      // Pass the memoryResource here because the libcudf semantics are that it should only be
+      // used to allocate memory for the columns that are returned.
+      memoryResource,
+      other: new_df,
+    });
+
+    old_df.drop(['categories']).dispose();
+    new_df.drop(['categories']).dispose();
```

Comment:
what's the `drop(...)` needed for?

Reply:
Dropping the `'categories'` Columns so we don't delete their memory.

Reply:
oh I see, a bit counterintuitive to first read